src/Pact/Native.hs

```diff
@@ -648,23 +653,35 @@ listModules i _ = do
   return $ toTermList tTyString $ map asString mods
 
 yield :: RNativeFun e
-yield i [t@(TObject (Object o _ _ _) _)] = do
-  eym <- use evalPactExec
-  case eym of
-    Nothing -> evalError' i "Yield not in defpact context"
-    Just {} -> do
-      o' <- enforcePactValue' o
-      (evalPactExec . _Just . peYield) .= Just o'
-      return t
-yield i as = argsError i as
+yield i as = case as of
+  [u@(TObject t _)] -> go "" t u
+  [(TLitString t), v@(TObject u _)] -> go (ChainId t) u v
+  _ -> argsError i as
+  where
+    go tid (Object o _ _ _) u = do
+      eym <- use evalPactExec
+      case eym of
+        Nothing -> evalError' i "Yield not in defpact context"
+        Just PactExec{..} -> do
+          o' <- enforcePactValue' o
+          e <- endorseM' i o' tid
```

Comment:
We need to capture the target chain also in the yield object as a `Maybe` -- given that this is true we can go to a `Maybe Endorsement` where the target chain is in the endorsement object, ie `Yield (ObjectMap PactValue) (Maybe Endorsement)` and Endorsement captures the other stuff.

Reply:
Also that way the "" value for `tid` can go away: if we have an endorsement we necc. have tid, txid whatever

Reply:
Okay, I'll address this today